internal/ioutil/ioutil.go

```diff
@@ -35,9 +34,9 @@ import (
 
 // Block sizes constant.
 const (
-	BlockSizeSmall       = 32 * humanize.KiByte // Default r/w block size for smaller objects.
-	BlockSizeLarge       = 2 * humanize.MiByte  // Default r/w block size for larger objects.
-	BlockSizeReallyLarge = 4 * humanize.MiByte  // Default write block size for objects per shard >= 64MiB
+	BlockSizeSmall       = 128 * humanize.KiByte // Default r/w block size for smaller objects.
+	BlockSizeLarge       = 2 * humanize.MiByte   // Default r/w block size for larger objects.
+	BlockSizeReallyLarge = 4 * humanize.MiByte   // Default write block size for objects per shard >= 64MiB
```

Comment:
I think we are writing too many small files for this to not cause too much overallocation.

So right now "small" is used for < 128K. So we would be looking at max 4 syscalls to write the file.

We can make the `BlockSizeLarge` cheaper and differentiate it more from "ReallyLarge":

```suggestion
	BlockSizeSmall       = 32 * humanize.KiByte // Default r/w block size for smaller objects.
	BlockSizeLarge       = 1 * humanize.MiByte   // Default r/w block size for larger objects.
	BlockSizeReallyLarge = 4 * humanize.MiByte   // Default write block size for big objects.
```

I would then suggest changing this: https://github.com/minio/minio/blob/88a89213ff6663f2d3ee1de7f3293497ab9844e8/cmd/xl-storage.go#L2115-L2126 to

```Go
	switch {
	case fileSize > 0 && fileSize >= xioutil.BlockSizeReallyLarge:
		// use a larger 4MiB buffer for a really large streams.
		bufp = xioutil.ODirectPoolXLarge.Get().(*[]byte)
		defer xioutil.ODirectPoolXLarge.Put(bufp)
	case fileSize <= xioutil.BlockSizeSmall:
		bufp = xioutil.ODirectPoolSmall.Get().(*[]byte)
		defer xioutil.ODirectPoolSmall.Put(bufp)
	default:
		bufp = xioutil.ODirectPoolLarge.Get().(*[]byte)
		defer xioutil.ODirectPoolLarge.Put(bufp)
	}
```

So everything up to 4MiB is written with a single syscall (plus DirectIO splitting).

We could set the switchover to `xioutil.BlockSizeReallyLarge*2` and `xioutil.BlockSizeSmall*2` to reduce to overalloc a bit - but from a pure performance point the above will probably be the best option.